### PR TITLE
Refactor EventModLookup and add tests

### PR DESCRIPTION
**Pull request by JonoPrest**
- moves all Js.Dict.get to optimized util
- Adds a field for quick lookup of wildcard events
- Adds test coverage of all the core functions in the module
Screenshot 2024-08-30 at 14 39 20

### CODE REVIEW

File: codegenerator/cli/templates/static/codegen/src/bindings/RescriptMocha.res [hunk 1]
```diff
@@ -20,7 +20,7 @@ module Assert = {
   @module("assert") external ifError: 'a => unit = "ifError"
 
   @module("assert")
-  external throws: (unit => 'a, ~error: Js.Exn.t, ~message: string=?) => unit = "throws"
+  external throws: (unit => 'a, ~error: Js.Exn.t=?, ~message: string=?) => unit = "throws"
```
Comment (JonoPrest) [inline codegenerator/cli/templates/static/codegen/src/bindings/RescriptMocha.res:23]: I think this is correct? I didn't actually want to pass in the expected error.

File: codegenerator/cli/templates/static/codegen/src/EventModLookup.res [hunk 2]
```diff
@@ -5,79 +5,86 @@ type errorKind = WildcardSighashCollision | Duplicate
...
     all: [],
     byContractName: Js.Dict.empty(),
   }
 
   let isWildcard = (eventMod: eventMod) => {
     let module(Event) = eventMod
-    Event.handlerRegister->Types.HandlerTypes.Register.getEventOptions->(v => v.Types.HandlerTypes.wildcard)
+    Event.handlerRegister
+    ->Types.HandlerTypes.Register.getEventOptions
+    ->(v => v.Types.HandlerTypes.wildcard)
```
Comment (DZakh) [inline codegenerator/cli/templates/static/codegen/src/EventModLookup.res:23]: This is the first time I see the syntax 😁

File: codegenerator/cli/templates/static/codegen/src/bindings/RescriptMocha.res (hunk 1 above)
Comment (DZakh) [reply codegenerator/cli/templates/static/codegen/src/bindings/RescriptMocha.res:23]: Yes, I think it's correct to have it optional. But the field is usually used to test error messages, which I think we should also do. But I'm not sure that the Js.Exn.t is correct here, for example in Ava.js I usually use you can pass a simple object with error assertions.

Comment (DZakh) [reply codegenerator/cli/templates/static/codegen/src/bindings/RescriptMocha.res:23]: Let's just type it `'error`

https://nodejs.org/api/assert.html#assertthrowsfn-error-message

File: codegenerator/cli/templates/static/codegen/src/bindings/RescriptMocha.res [hunk 3]
```diff
@@ -20,7 +20,7 @@ module Assert = {
...
   @module("assert")
   external doesNotThrow: (unit => 'a, ~error: Js.Exn.t, ~message: string=?) => unit = "doesNotThrow"
```
Comment (DZakh) [inline codegenerator/cli/templates/static/codegen/src/bindings/RescriptMocha.res:25]:
```suggestion
  external doesNotThrow: (unit => 'a, ~error: 'error=?, ~message: string=?) => unit = "doesNotThrow"
```

File: scenarios/test_codegen/test/lib_tests/EventModLookup_test.res [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        ~isWildcard=false,
+      )->toInternal
+
+    [EventModLookup.Duplicate, WildcardSighashCollision]->Belt.Array.forEach(
+      errorKind => {
+        Assert.throws(
+          () => {
+            Error({eventMod, errorKind})->EventModLookup.unwrapAddEventResponse(~chain=mockChain)
+          },
+        )
```
Comment (DZakh) [inline scenarios/test_codegen/test/lib_tests/EventModLookup_test.res:233]:
```suggestion
        Assert.throws(
          () => {
            Error({eventMod, errorKind})->EventModLookup.unwrapAddEventResponse(~chain=mockChain)
          },
          ~error={
            "message": "I don't know what, but it should be tested"
          }
        )
```

Comment (DZakh) [reply scenarios/test_codegen/test/lib_tests/EventModLookup_test.res:233]: Also, it'll require to remove forEach

**Review by DZakh — APPROVED** [review]
Looks very good. And having tests for error messages will make it even better :)

File: codegenerator/cli/templates/static/codegen/src/EventModLookup.res (hunk 2 above)
Comment (JonoPrest) [reply codegenerator/cli/templates/static/codegen/src/EventModLookup.res:23]: Haha, I don't really have too strong an opinion but thought it flows better than using round brackets around the whole statement and accessing .wildcard at the end. I might use it again.

**Pull request merged**